[IMP] carousel: export in xlsx

hokolomopo opened the pull request:
wip 

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

hokolomopo committed:
[IMP] model: improve figure export data

Currently, charts/images/carousels are exported inside `figure.data`
in the JSON. That's not ideal, `figure.data` then needs to be typed as
`any`, and the charts plugin need to be aware of carousels for its
import.

This commit changes the export, now each plugin has its own key in the
data, so it can be typed correctly and they are not interlinked.

Task: 5002886

hokolomopo committed:
[IMP] carousel: add xlsx export

This commit adds the export of carousel to the xlsx. As Excel does not
support carousels, we import each chart of the carousel individually
into separate figures.

Task: 5056097

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/o-spreadsheet/pull/7055.png)](https://mergebot.odoo.com/odoo/o-spreadsheet/pull/7055)

hokolomopo committed:
[FIX] tests: xlsx fix image ids at export

The conversion imageId <=> xlsx image id was done with a global variable
which was not reset between tests, leading to test giving different
result when running the whole file or a single test.

Task: 5056097
